apps/dav/lib/CalDAV/CalendarImpl.php

```
@@ -145,11 +145,16 @@ public function createFromString(string $name, string $calendarData): void {
 		// so set the custom principal here
 		$plugin->setCurrentPrincipal($this->calendar->getPrincipalURI());
 
+		// Build full calendar path
+		$parts = explode('/',  $this->calendar->getPrincipalURI());
+		$user = $parts[2];
+		$fullCalendarFilename = sprintf('calendars/%s/%s/%s', $user, $this->calendarInfo['uri'], $name);
```

I'd add a check to see if `$this->calendarInfo['uri']` is defined.

---

Fair point but if it's missing we can only fail hard, right?

---

Yup, exception is good.